dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1411,49 +1412,70 @@ public void patchCertificateNonAdminUser() throws Exception {
 
         context.restoreAuthSystemState();
 
         String token = getAuthToken(admin.getEmail(), password);
 
-        String newPassword = "newpass";
-
-        List<Operation> ops = new ArrayList<Operation>();
-        ReplaceOperation replaceOperation = new ReplaceOperation("/password", newPassword);
-        ops.add(replaceOperation);
+        List<Operation> ops = new ArrayList<>();
+        AddOperation addOperation = new AddOperation("/password", null);
+        ops.add(addOperation);
         String patchBody = getPatchContent(ops);
 
-        // initialize passwd
+        // adding null pw should return bad request
         getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
-                .content(patchBody)
-                .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+            .content(patchBody)
+            .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                        .andExpect(status().isBadRequest());
+
+        // login with original password
+        token = getAuthToken(ePerson.getEmail(), originalPw);
+        getClient(token).perform(get("/api/"))
                         .andExpect(status().isOk());
 
+    }
 
-        List<Operation> ops2 = new ArrayList<Operation>();
-        ReplaceOperation replaceOperation2 = new ReplaceOperation("/password", null);
-        ops2.add(replaceOperation2);
-        patchBody = getPatchContent(ops2);
+    @Test
+    public void patchPasswordNotInitialised() throws Exception {
 
-        // should return bad request
+        context.turnOffAuthorisationSystem();
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("userNotInitialised@example.com")
+                                        .withCanLogin(true)
+                                        .build();
+
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        String newPassword = "newpass";
+
+        List<Operation> ops = new ArrayList<Operation>();
+        AddOperation addOperation = new AddOperation("/password", newPassword);
+        ops.add(addOperation);
+        String patchBody = getPatchContent(ops);
+
+        // initialize password with add operation, not set during creation
         getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
                 .content(patchBody)
                 .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
-                        .andExpect(status().isBadRequest());
+                        .andExpect(status().isOk());
 
-        // login with original password
+        // login with new password => succeeds
         token = getAuthToken(ePerson.getEmail(), newPassword);
         getClient(token).perform(get("/api/"))
                         .andExpect(status().isOk());
```

Comment:
I have approved this PR but I want to note that this test is not exactly what I would expect. It would be better to test for the authn/status endpoint to verify that we are really loggedin. It is not documented what should happen trying to access a public endpoint with an invalid token. @tdonohue do you agree? should we open a ticket for that? again this PR can be merged as is as this is unrelated with the PR

Reply:
@Raf-atmire : @abollini is correct (and I overlooked this).  I don't think this test is correct (and I see you've reused this same test in your changes to `patchPasswordMissingValue()`).

Instead of doing a GET against `/api/`, we should be doing a GET against `/api/authn/status` and checking that it returns "authenticated=true". So, what you should be doing here is replace these two lines (starting with `getClient(` with somethin like this: https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthenticationRestControllerIT.java#L211-L216

That'd be a more accurate test to verify that the login *works*.  I'm actually not sure if running a GET against `/api/` even requires authentication or not -- in fact based on the response of the demo REST API, it seems like that request would NOT require authentication: https://dspace7.4science.cloud/server/api/

So, if you could fix this check here and on line 1440 above, then this PR should be ready to merge.  Thanks!